odoo/addons/base/wizard/base_language_install.py

```diff
@@ -9,56 +9,37 @@ class BaseLanguageInstall(models.TransientModel):
     _description = "Install Language"
 
     @api.model
-    def _default_language(self):
+    def _default_lang_ids(self):
         """ Display the selected language when using the 'Update Terms' action
             from the language list view
         """
         if self._context.get('active_model') == 'res.lang':
-            lang = self.env['res.lang'].browse(self._context.get('active_id'))
-            return lang.code
+            return self.env['res.lang'].browse(self._context.get('active_ids'))
         return False
 
-    @api.model
-    def _get_languages(self):
-        return [[code, name] for code, _, name, *_ in self.env['res.lang'].get_available()]
-
-    lang = fields.Selection(_get_languages, string='Language', required=True,
-                            default=_default_language)
+    lang_ids = fields.Many2many('res.lang', 'res_lang_install_rel', 'language_wizard_id', 'lang_id', 'Languages', default=_default_lang_ids)
     overwrite = fields.Boolean('Overwrite Existing Terms',
                                default=True,
                                help="If you check this box, your customized translations will be overwritten and replaced by the official ones.")
-    state = fields.Selection([('init', 'init'), ('done', 'done')],
-                             string='Status', readonly=True, default='init')
 
     def lang_install(self):
         self.ensure_one()
         mods = self.env['ir.module.module'].search([('state', '=', 'installed')])
-        self.env['res.lang']._activate_lang(self.lang)
-        mods._update_translations(self.lang, self.overwrite)
-        self.state = 'done'
+        for lang in self.with_context(active_test=False).lang_ids:
+            lang.active = True
```

Comment:
Why not calling ``_activate_lang`` as before ? If a custom behavior is implemented on activate, maybe we should still call it to keep it like before...

Reply:
Because I would still need the loop, right? 
```py
for lang in self.with_context(active_test=False).lang_ids:
    self.env['res.lang']._activate_lang(lang.code) # instead of lang.active = True
    mods._update_translations(lang.code, self.overwrite)
```

So it would only have the effect of doing `lang.active = True`, and would be quite pointless to call another function for that no?

Reply:
The loop is needed for `_update_translations`

Reply:
it looks like `_update_translations` can be called in batch (by giving a list / tuple instead of a string), can't you do something like this ?
```python
codes = self.with_context(active_test=False).lang_ids.mapped('code')
mods._update_translations(codes, self.overwrite)
```

![screenshot_spect](https://user-images.githubusercontent.com/54316226/149524080-c492fcba-0940-4bc3-95f3-bb1a03fcd556.png)

For `_activate_lang`, it's ok for me to do a simple 
```python
self.with_context(active_test=False).lang_ids.active = True
```

So you might not need the loop

Reply:
Done :)